File: server/ruby/app/infrastructure/mappers/payment_details/customer_device_output_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class CustomerDeviceOutputMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::CustomerDeviceOutput.new
          dto.ip_address_country_code = response.ip_address_country_code

          dto
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/payment_details/customer_device_output_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.customer_device_output import CustomerDeviceOutput
from onlinepayments.sdk.domain.customer_device_output import CustomerDeviceOutput as CustomerDeviceOutputSdk

class CustomerDeviceOutputMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[CustomerDeviceOutputSdk]) -> Optional[CustomerDeviceOutput]:
        if response is None:
            return None

        return CustomerDeviceOutput(
            ip_address_country_code = response.ip_address_country_code
        )